pyperf/_utils.py

```diff
@@ -317,9 +320,34 @@ def __exit__(self, *args):
 class ReadPipe(_Pipe):
     def open_text(self):
         file = open(self._fd, "r", encoding="utf8")
+        if self._timeout:
+            os.set_blocking(file.fileno(), False)
         self._file = file
         return file
 
+    def read_text(self):
+        with self.open_text() as rfile:
+            if self._timeout:
+              return self._read_text_timeout(rfile, self._timeout)
+            else:
+                return rfile.read()
+
+    def _read_text_timeout(self, rfile, timeout):
+        start_time = time.time()
```

Comment:
Can you please use time.monotonic() instead? It's not affected by DST or sys admin changing manually the system clock.